### PR TITLE
Allow attaching path

### LINKED ISSUE #1506: Allow attaching a path
It's common that users pass a file path (usually for a png or jpg file) as the first argument to the `attach` method. This leads to attachments that cannot displayed. One of the reasons users attach paths is that tools like Capybara's `save_screenshot` method returns a path instead of image data. Many users don't realise they have to `IO.read` the path before passing it to `attach`.

I think the `attach` method should behave as follows:

If the argument is a string, and the file exists, read the file and attach that.
Use a file extension to media type lookup to determine the media type. If a media type was specified, throw an error if the specified media type is different from the inferred one.

### PR DESCRIPTION
**Pull request by aurelien-reeves**
**Is your pull request related to a problem? Please describe.**

Fixes #1506

**Describe the solution you have implemented**

We plan to allow attaching files by specifying their path
Cucumber will take care of opening and reading it.

### CODE REVIEW

**Comment by luke-hill** [comment]
Make sure to add a path to a file auto-detects (Maybe based off extension??)

You could have an outline with
```
foo.png
bar.txt
baz.md
bay.jpeg
```
e.t.c.

**Comment by aurelien-reeves** [comment]
> 
> 
> Make sure to add a path to a file auto-detects (Maybe based off extension??)
> 
> You could have an outline with
> 
> ```
> foo.png
> bar.txt
> baz.md
> bay.jpeg
> ```
> 
> e.t.c.

Yeah, that is interesting indeed 👍 
At first I'll push a basic feature file with no auto-detection specified for a first iteration. Once it passes, we will be able to iterate on that auto-detection ideas

**Comment by aslakhellesoy** [comment]
This looks great @aurelien-reeves! Nice and simple.

Reading your code I had an idea that we might be able to add at the same time: Make the media type optional when you're passing an existing path. We could use a [library](https://github.com/mime-types/ruby-mime-types) to look up a media type for a file extension.

File: lib/cucumber/glue/proto_world.rb [hunk 1]
```diff
@@ -111,8 +112,13 @@ def log(*messages)
         messages.each { |message| attach(message.to_s.dup, 'text/x.cucumber.log+plain') }
       end
 
-      def attach(file, media_type)
-        super
+      def attach(file, media_type = nil)
+        return super unless File.file?(file)
+
+        content = File.open(file, 'rb', &:read)
```
Comment (olleolleolle) [inline lib/cucumber/glue/proto_world.rb:118]: Would it be possible to use this here?

```suggestion
        content = File.read(file, mode: 'rb')
```

https://rubyapi.org/2.5/o/io#method-c-read

Comment (aurelien-reeves) [reply lib/cucumber/glue/proto_world.rb:118]: I think so, yes 👍

**Pull request merged**